Merge front_responsive to master

MatiSalinas opened the pull request:
### Pull Request Description

#### What does this PR do?
This pull request addresses the responsiveness of the frontend and refactors the FormLab component to use `react-hook-form` for better form management. It includes the following changes:

- Disables animations on the News component for mobile devices.
- Adds media queries for devices with a screen width of 800px or less, optimizing the layout for smaller screens.
- Transforms the navbar into a mobile-friendly menu using the `isMobile` flag to provide different layouts for mobile and desktop views.
- Rewrites the FormLab component to use `react-hook-form` for handling form data and integrates `zod` for form validation.

#### Changes Made:
1. **Animations in News Component:**
   - Introduced a flag (`isMobile`) to disable `framer-motion` animations on mobile devices, improving the user experience on smaller screens.
  
2. **Mobile Responsiveness:**
   - Added media queries targeting screens 800px and below.
   - Adjusted the layout by switching from `flex` to `column` for most elements and setting component widths to `100vw`.

3. **Navbar Refactor:**
   - Used the `isMobile` flag to provide a responsive menu, replacing the traditional navbar with a collapsible menu on mobile devices.

4. **FormLab Refactor:**
   - Replaced the form state management with `react-hook-form`'s `register` method.
   - Integrated `zod` for schema-based validation, ensuring consistent validation and error handling.



#### Related Issues:
- [Link to related issue #7](https://github.com/Esc-IngComp-UNC/computer-laboratory-web-page/issues/7)

MatiSalinas committed:
React proyect created

MatiSalinas committed:
NavBar component created

MatiSalinas committed:
Carousel added

MatiSalinas committed:
About us component added

MatiSalinas committed:
News component prototype added

MatiSalinas committed:
fixed margin errors for smaller displays(laptops)

MatiSalinas committed:
comments, .gitignore file moved to root folder, .gitkeep, document format

MatiSalinas committed:
Faq component Created

MatiSalinas committed:
Form component added

MatiSalinas committed:
Map and contact components added

MatiSalinas committed:
Relative path @ and endlines added.

prietojulii committed:
Merge pull request #5 from MatiSalinas/master

Merge to intermediate branch

MatiSalinas committed:
Media query added to devices with a max-width of 600px

MatiSalinas committed:
Disabled animations on mobile News component now uses variants for animations

MatiSalinas committed:
Faq now uses variants and it works as intended on mobile

MatiSalinas committed:
Contact component now is fully responsive

MatiSalinas committed:
form now uses react-hook-form and zod to validate its fields

MatiSalinas committed:
navBar menu on mobile and scroll into view

MatiSalinas committed:
missing validation

MatiSalinas committed:
collapsible menu for mobile

prietojulii commented:
Great job, Mat! I only have a few comments

prietojulii commented on web/src/App.css:
```diff
@@ -0,0 +1,679 @@
+  align-items: center;
+  justify-content: space-between;
+  color: #017481;
+  background-color: #f2f3f4;
```
We can use CSS variables to make maintenance easier. For example, colors could be placed in variables and reused

prietojulii commented on web/src/App.jsx:
```diff
@@ -0,0 +1,37 @@
+      <div className="container">
+        <NavBar></NavBar>
+        <CarouselImages></CarouselImages>
+        <Element name="aboutUs">
```
The words displayed to the user must be in Spanish

prietojulii replied:
Maintain the order of the menu with the presentation order.
1- aboutus
2- novedades
3- faq
4-contacto

prietojulii commented on web/src/App.jsx:
```diff
@@ -0,0 +1,37 @@
+        <Element name="novedades">
+        <News></News>
+        </Element>
+        <Element name="faq">
```
the same here

prietojulii commented on web/src/components/Contact.jsx:
```diff
@@ -0,0 +1,42 @@
+import twitter from '/svgs/twitterx.svg'
+
+function Contact(){
+    //TODO add SVG to each link
```
This comment can be deleted?

prietojulii replied:
```suggestion
```

prietojulii commented on web/.gitkeep:
this file can be deleted

prietojulii commented on web/index.html:
```diff
@@ -0,0 +1,12 @@
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Laboratorio Computacion</title>
```
```suggestion
    <title>Laboratorio de Computación</title>
```

prietojulii commented on web/src/components/Contact.jsx:
```diff
@@ -0,0 +1,42 @@
+import youtube from '/svgs/youtube.svg'
+import instagram from '/svgs/instagram.svg'
+import linkedin from '/svgs/linkedin.svg'
+import twitter from '/svgs/twitterx.svg'
```
We havent twitter or youtube. Can you delete them?

prietojulii commented on web/src/components/Contact.jsx:
```diff
@@ -0,0 +1,42 @@
+            <p>Laboratorio de Computación:</p>
+
+            <p>Correo: lc.fcefyn@gmail.com</p>
+            <p>Instagram: c.fcefyn</p>
```
```suggestion
            <div className="socialMedia">
                <img src={instagram} alt="InstagramLogo" />
            <a target='_blank' href='https://www.instagram.com/lc.fcefyn/'>Instagram</a>
            </div>
```

prietojulii commented on web/src/components/Contact.jsx:
```diff
@@ -0,0 +1,42 @@
+            <div className="infoL">
+            <p>Laboratorio de Computación:</p>
+
+            <p>Correo: lc.fcefyn@gmail.com</p>
```
Can we use an image/icon for GMAIL. For example https://upload.wikimedia.org/wikipedia/commons/7/7e/Gmail_icon_%282020%29.svg

prietojulii commented on web/src/components/MapGoogle.jsx:
```diff
@@ -0,0 +1,12 @@
+    <div className="mapa" >
+        <div className="borderT"></div>
+
+        <div className="mapa"><iframe src="https://www.google.com/maps/embed?pb=!1m10!1m8!1m3!1d13617.083057234764!2d-64.181348!3d-31.434212!3m2!1i1024!2i768!4f13.1!5e0!3m2!1ses-419!2sar!4v1727935597767!5m2!1ses-419!2sar" width={'100%'}  height={600} style={{border:0}} allowFullScreen={''} loading={"lazy"} referrerPolicy={"no-referrer-when-downgrade"}></iframe></div>
```
```suggestion
        <div className="mapa"><iframe src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13617.083057234764!2d-64.181348!3d-31.434212!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x9432a2f54d372bf1%3A0xf1d83ddc363cda16!2sFacultad%20de%20Ciencias%20Exactas%2C%20F%C3%ADsicas%20y%20Naturales!5e0!3m2!1ses-419!2sar!4v1731029204729!5m2!1ses-419!2sar" width={'100%'}  height={600} style={{border:0}} allowFullScreen={''} loading={"lazy"} referrerPolicy={"no-referrer-when-downgrade"}></iframe></div>
```

prietojulii replied:
Esto marca un punto rojo en la ubicación

prietojulii commented on web/src/components/Contact.jsx:
```diff
@@ -0,0 +1,42 @@
+
+            <div className="socialMedia">
+                <img src={twitter} alt="twitterXLogo" />    
+            <a target='_blank' href='https://twitter.com/FCEFyN_UNC' >twitter</a>
```
```suggestion
            <a target='_blank' href='https://twitter.com/fcefyn_unc' >twitter</a>
```

MatiSalinas committed:
customHook for mobile, css variables, and more

MatiSalinas committed:
updated readme with setup instructions

MatiSalinas commented:
Thanks for your feedback, I made the suggested changes in the last commit [0aa27ba](https://github.com/Esc-IngComp-UNC/computer-laboratory-web-page/pull/8/commits/0aa27ba183043bea8ecdc585a149fdea1c351891).

prietojulii commented:
It's amazing! Good Job!

GabrielEValenzuela commented on web/src/main.jsx:
```diff
@@ -0,0 +1,9 @@
+import { StrictMode } from 'react'
+import { createRoot } from 'react-dom/client'
+import App from './App.jsx'
```
```suggestion
import App from '@/src/App.jsx'
```

GabrielEValenzuela reviewed (changes requested):
Great work Mati! 

Please check missing endlines and one little improvement on the path

MatiSalinas committed:
app path uses @ and endline

MatiSalinas commented:
added endline and changed path on [27635ab](https://github.com/Esc-IngComp-UNC/computer-laboratory-web-page/commit/27635aba63d92699104e06b16fba4c6528c3aa9f).

GabrielEValenzuela reviewed (approved):
LGTM
